AWSIoTPythonSDK/core/protocol/connection/cores.py

```diff
@@ -482,11 +482,11 @@ def _verifyWSSAcceptKey(self, srcAcceptKey, clientKey):
 
     def _handShake(self, hostAddress, portNumber):
         CRLF = "\r\n"
-        IOT_ENDPOINT_PATTERN = r"^[0-9a-zA-Z]+\.iot\.(.*)\.amazonaws\..*"
+        IOT_ENDPOINT_PATTERN = r"^[0-9a-zA-Z]+(\-ats)*\.iot\.(.*)\.amazonaws\..*"
```

Comment:
I think if you replace `(\-ats)*` with `(?:\-ats)?` you'll a) only allow 0 or 1 instance of `-ats` and b) not capture `-ats`, since you don't actually need it.

Reply:
thank you for reply
I fixed regexp pattern